### PR TITLE
ARROW-17892: [CI] Use Python 3.10 in AppVeyor build

### PR DESCRIPTION
**Pull request by AlenkaF**
This PR tries to upgrade Python version used in `AppVeyor` setup and removes `CONDA_DLL_SEARCH_MODIFICATION_ENABLE` as it should not be needed with Python 3.10.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
https://github.com/apache/arrow/pull/13552 is fixing the cython tests, required the get this passing

**Comment by jorisvandenbossche** [comment]
@AlenkaF https://github.com/apache/arrow/pull/13552 is merged, so you an update this one I think

File: appveyor.yml [hunk 1]
```diff
@@ -57,4 +55,6 @@ build_script:
 test: off
 
 after_build:
-  - ccache -s
+  # Try to get the info about the cache directory
+  - ccache -s -p
+  - sccache -s -p
```
Comment (jorisvandenbossche) [inline appveyor.yml:60]: I suppose sccache is not installed here

**Comment by jorisvandenbossche** [comment]
It worked: `(default) cache_dir = C:\Users\appveyor\AppData\Local\ccache`

File: appveyor.yml (hunk 1 above)
Comment (AlenkaF) [reply appveyor.yml:60]: Yeah, not sure what magic I had in mind :D 
Planned to remove with the `-p` flag (or should we keep it?)

Comment (jorisvandenbossche) [reply appveyor.yml:60]: It doesn't hurt to keep it I think

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Appveyor is passing, so merging. Thanks!

**Pull request merged**